test/secmemtest.c

```diff
@@ -75,12 +75,31 @@ static int test_sec_mem(void)
     TEST_ptr_null(OPENSSL_secure_malloc((size_t)-1));
     TEST_true(CRYPTO_secure_malloc_done());
 
+    /*-
+     * There was also a possible infinite loop when the number of
+     * elements was 1<<31, as |int i| was set to that, which is a
+     * negative number. However, it requires minimum input values:
+     *
+     * CRYPTO_secure_malloc_init((size_t)1<<34, (size_t)1<<4);
+     *
+     * Which really only works on 64-bit systems, and even then the
+     * code attempts to allocate 16 GB secure memory arena. Linux
+     * can deal with this better than other Unixy OS's (e.g. MacOS)
```

Comment:
> I have not seen it fail after 64KB on Linux

Ah! It says limit is ignored if process has CAP_IPC_LOCK capability/privilege on... Root processes usually get all capabilities. So it was wrong of me to say that limit is enforced for root.

> I would have to confirm.

It should fail if application drops privileges before calling CRYPTO_secure_malloc_init.

> IMHO, the application would have to raise any limits.

Maybe it would be appropriate to say few words in doc?